src/modules/newDeploy.ts

```diff
@@ -0,0 +1,88 @@
+import chalk from 'chalk'
+import { Messages } from '../lib/constants/Messages'
+import {
+  createRegistryClient,
+  parseLocator,
+  logger,
+  ManifestEditor,
+  ManifestValidator,
+  promptConfirm,
+  SessionManager,
+  returnToPreviousAccount,
+  switchAccount,
+} from 'vtex'
+
+const switchToVendorMessage = (vendor: string): string => {
+  return `You are trying to deploy this app in an account that differs from the indicated vendor. Do you want to deploy in account ${chalk.blue(
+    vendor
+  )}?`
+}
+
+const promptDeploy = (app: string) => promptConfirm(`Are you sure you want to deploy app ${app}`)
+
+const deployRelease = async (app: string, force: boolean): Promise<boolean> => {
+  const { vendor, name, version } = parseLocator(app)
+  const session = SessionManager.getSingleton()
+
+  if (vendor !== session.account) {
+    const canSwitchToVendor = await promptConfirm(switchToVendorMessage(vendor))
+
+    if (!canSwitchToVendor) {
+      return false
+    }
+
+    await switchAccount(vendor, {})
+  }
+
+  const context = { account: vendor, workspace: 'master', authToken: session.token }
+  const registry = createRegistryClient(context)
+  const path = `${vendor}.${name}?ignoreWaitPeriod=${force}`
+
+  await registry.validateApp(path, version)
```

Comment:
**nit** WDYT about putting this in a separate helper?

Reply:
Sure